packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -134,23 +134,44 @@ function createResponseEvent(cfnRequest: AWSLambda.CloudFormationCustomResourceE
   //
   // validate that onEventResult always includes a PhysicalResourceId
 
-  if (!onEventResult || !onEventResult.PhysicalResourceId) {
-    throw new Error(`onEvent response must include a PhysicalResourceId for all request types`);
-  }
+  onEventResult = onEventResult || { };
+
+  // if physical ID is not returned, we have some defaults for you based
+  // on the request type.
+  const physicalResourceId = onEventResult.PhysicalResourceId || defaultPhysicalResourceId(cfnRequest);
 
   // if we are in DELETE and physical ID was changed, it's an error.
-  if (cfnRequest.RequestType === 'Delete' && onEventResult.PhysicalResourceId !== cfnRequest.PhysicalResourceId) {
+  if (cfnRequest.RequestType === 'Delete' && physicalResourceId !== cfnRequest.PhysicalResourceId) {
     throw new Error(`DELETE: cannot change the physical resource ID from "${cfnRequest.PhysicalResourceId}" to "${onEventResult.PhysicalResourceId}" during deletion`);
   }
 
   // if we are in UPDATE and physical ID was changed, it's a replacement (just log)
-  if (cfnRequest.RequestType === 'Update' && onEventResult.PhysicalResourceId !== cfnRequest.PhysicalResourceId) {
+  if (cfnRequest.RequestType === 'Update' && physicalResourceId !== cfnRequest.PhysicalResourceId) {
     log(`UPDATE: changing physical resource ID from "${cfnRequest.PhysicalResourceId}" to "${onEventResult.PhysicalResourceId}"`);
   }
 
   // merge request event and result event (result prevails).
   return {
     ...cfnRequest,
-    ...onEventResult
+    ...onEventResult,
+    PhysicalResourceId: physicalResourceId,
   };
+}
+
+/**
+ * Calculates the default physical resource ID based in case user handler did
+ * not return a PhysicalResourceId.
+ *
+ * For "CREATE", it uses the RequestId.
+ * For "UPDATE" and "DELETE" and returns the current PhysicalResourceId (the one provided in `event`).
+ */
+function defaultPhysicalResourceId(req: AWSLambda.CloudFormationCustomResourceEvent): string {
+  switch (req.RequestType) {
+    case 'Create':
+      return req.RequestId;
+
+    case 'Update':
+    case 'Delete':
```

Comment:
Throw in a `default:` clause? (That either returns `req.PhysicalResourceId` or throws)

Reply:
ok, added a default with an exception